prompting/llms/vllm_llm.py

```diff
@@ -81,7 +82,8 @@ def __call__(self, composed_prompt: str, **model_kwargs: Dict) -> str:
         sampling_params = SamplingParams(
             temperature=temperature, top_p=top_p, max_tokens=max_tokens
         )
-        output = self.llm.generate(composed_prompt, sampling_params, use_tqdm=True)
+        with self._LOCK:
```

Comment:
you're defining 

_LOCK (no self.) but then are using with self._LOCK. Is that correct?

Reply:
Yeah, it is defined as a class member to ensure that the LLM/GPU is locked for only one process at a time across the project